lib/classes/collection.php

```diff
@@ -188,52 +188,73 @@ public static function getByBaseId(Application $app, $base_id)
             ));
         }
 
-        $repository = self::getRepository($app, $reference->getDataboxId());
-        $collection = $repository->find($reference->getCollectionId());
+        return self::getAvailableCollection($app, $reference->getDataboxId(), $reference->getCollectionId());
+    }
 
-        if (!$collection) {
-            throw new Exception_Databox_CollectionNotFound(sprintf(
-                "Collection with base_id %s could not be found",
-                $base_id
-            ));
-        }
+    /**
+     * @param  Application $app
+     * @param  databox $databox
+     * @param  int $collectionId
+     * @return collection
+     */
+    public static function getByCollectionId(Application $app, databox $databox, $collectionId)
+    {
+        assert(is_int($collectionId));
```

Comment:
Why not use the assertions provided by beberlei since we use already use them in the code base ?